### PR TITLE
feat(content): rebuilding gunstores and spawn changes

### PR DESCRIPTION
**Pull request by Lamandus**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change
Gun stores used a old way to distribute items. Also the regular one looked a bit... ugly. This should make it a bit better. Changes of spawns are also included.

Guns will always spawn in gun displays.
Magazines and gun mods will always spawn in racks.
Ammo can be found only in the vending machine or in the displays

## Describe the solution
1. removed the location based spawns.
2. add furniture based spawns
3. made the gun store a bit bigger, removed parking spaces.
4. added ATM to the gun store outside
5. added dumpsters to the gun stores.
6. Display gun spawns more rifles now.
7. remove guns and magazines from ammo vending machine
8. spawn vending machines only at shooting ranges.

## Additional context
Before.
grafik
After:
grafik

### CODE REVIEW

**Comment by chaosvolt** [comment]
> Ammo can be found only in the vending machine.

Wait WHAT!? No.

Given how common we've made ammo available via random drops from nested gun spawns on zombies, making gun stores never have ammo on shelves, or at minimum locked display cases, renders them completely worthless as a location to even consider looting.

**Comment by Lamandus** [comment]
> > Ammo can be found only in the vending machine.
> 
> Wait WHAT!? No.
> 
> Given how common we've made ammo available via random drops from nested gun spawns on zombies, making gun stores never have ammo on shelves, or at minimum locked display cases, renders them completely worthless as a location to even consider looting.

Why, there are enough ammo to buy in these vending machines. There are up to 6 per shop, and ammo is dirt cheap.

**Comment by chaosvolt** [comment]
> Why, there are enough ammo to buy in these vending machines. There are up to 6 per shop, and ammo is dirt cheap.

Did you make sure that each shop variant even has a vending machine, if so? I'm very leery of that change to be honest.

**Comment by Lamandus** [comment]
> > Why, there are enough ammo to buy in these vending machines. There are up to 6 per shop, and ammo is dirt cheap.
> 
> Did you make sure that each shop variant even has a vending machine, if so? I'm very leery of that change to be honest.

of course! All of these (except those which are sheltered) have them. The sheltered ones weren't touched.

**Comment by Lamandus** [comment]
wait, I may found an error ^^

**Review by RoyalFox2140 — CHANGES_REQUESTED** [review]
You can't just upend the entire gameplay loop by putting all ammo in vending machines. That's gotta go.

You've also completely removed the area for a cashier to work but that's a fairly minor complaint, I'd probably move the display cases to form a cashier area because the shop looks bad without one.

**Review by chaosvolt — CHANGES_REQUESTED** [review]
I figured Royal would likely be the one to most strongly object to this yeah. My thoughts on this are mainly that twofold:
1. Gameplay-wise, I can't really trust that the vending machines are a good idea balance-wise. In fact there are multiple deep flaws with making ammo vending machines the sole method of getting ammo in gun stores. For starters, it locks the player into only being able to get ammo from gun stores via a single specific way. This in turn diminishes the value the one gun store where this was a novel way to get a reliable source of ammo via an uncommon method. It will also make it visually hard to gauge how useful the store is as a source of ammo, as the only ammo spawns are concentrated inside sealed containers.
2. Minor concerns include that it's unrealistic for all gun stores, especially in New England, to have zero display cases of ammo whatsoever, and that it diminishes the novelty value of the one variant where these are supposed to spawn.

In short:
1. Remove the gun vending machines from all variants except the one they're supposed to be in.
2. Place the ammo spawns in locked display cases instead, that already provides some level of risk since smashing them open will trigger an alarm and eyebot event, while still making it easy to tell at a glance that the store actually has ammo in it.
3. Ideally also bring back a dedicated cashier area. Parking would be nice if there's space for it but understandable if you can't set aside room for that.

Optional for the future: we oughta be able to try and lockpick open display cases. Reinforced display cases that require modestly hefty weapons to crack open but can also be pried or lockpicked could also be attempted someday (I have a few ideas where else those could be used too).

**Comment by RoyalFox2140** [comment]
I don't mean to be hard on anyone, but I'm also not eloquent enough to fully explain the gravity of how much this is changing.

**Comment by Tefnut** [comment]
I can see some reasons why we wouldn't want every gun store to use vending machines.

1. Not every gun store does that in the real world, and while realism isn't our wheelhouse, we still care slightly about it. It would be weird in all but the most high tech of modlists for there to be no gun stores that sell ammo the "old fashioned" way. Especially rarer ammo.
2. It makes gun stores worthless to loot without a pickaxe or cash card, things players either don't have until endgame, or just don't bother picking up usually.

I'm not against making ammo vendors more common, I am against making them the ONLY source of ammo in gun stores, as that will actively screw up several players' early-game playstyle.

Being as constructive as I can here, as gun stores DO seriously need some refinement, they've been basically untouched since we split from DDA.

**Comment by RoyalFox2140** [comment]
@chaosvolt In the future I wonder if we could set money to be a global resource you automatically loot without consuming turns and doesn't take inventory space, that way players who don't generally collect it still grab it. Cash cards would still exist but just get drained of cents automatically.

**Comment by chaosvolt** [comment]
This still has the vending machines in the other gun store variants, and as noted the cashier area has been pretty much nuked. Already seeing the first two variants look a lot more bland and same-y because of this:
image

**Comment by Tefnut** [comment]
> This still has the vending machines in the other gun store variants, and as noted the cashier area has been pretty much nuked. Already seeing the first two variants look a lot more bland and same-y because of this: image

with the first one there, I honestly like that they replaced the lockers with display cases, but not the rest of the redesign. The display cases would it nicer to glance into the back of the store and see what fun, exotic weapons i could work with.

The windows in the back, though, would mess up what is, in my opinion one of the best parts of the classic layout, that being having a nice shaded back area to hide from the dead in during a run. It also made sense to have no windows and be barred in, as it was the back room where they kept the really cool, expensive stuff.

**Comment by Lamandus** [comment]
> You can't just upend the entire gameplay loop by putting all ammo in vending machines. That's gotta go.
> 
> You've also completely removed the area for a cashier to work but that's a fairly minor complaint, I'd probably move the display cases to form a cashier area because the shop looks bad without one.

grafik
this is meant to be the cashier area but it isn't too clear, maybe removing the windows will help

**Comment by RoyalFox2140** [comment]
It would also be more ideal if display cases could be lockpicked and vending machines hacked, as anyone who doesn't realize they need to gather money and isn't expecting to need a jackhammer is just out of luck. Those are both code projects for another day.

The other big problem right now with cash is having to find an ATM to load the dozens of cash cards you've looted onto a single card, which isn't ideal gameplay.

**Comment by Lamandus** [comment]
> It would also be more ideal if display cases could be lockpicked and vending machines hacked, as anyone who doesn't realize they need to gather money and isn't expecting to need a jackhammer is just out of luck. Those are both code projects for another day.
> 
> The other big problem right now with cash is having to find an ATM to load the dozens of cash cards you've looted onto a single card, which isn't ideal gameplay.

but the gun store now has an ATM. And you can carry just every cash card with you, they stack together.

**Comment by chaosvolt** [comment]
Either way, please actually fix the layouts. I like the use of display cases but it definitely needs some sanity-checking, and for the vending machines to be removed from the normal store variants.

**Comment by Tefnut** [comment]
> > You can't just upend the entire gameplay loop by putting all ammo in vending machines. That's gotta go.
> > You've also completely removed the area for a cashier to work but that's a fairly minor complaint, I'd probably move the display cases to form a cashier area because the shop looks bad without one.
> 
> grafik this is meant to be the cashier area but it isn't too clear, maybe removing the windows will help

I suppose that makes sense, though the old "Cashier cage" is a classic thing that was really an identifying feature of the layout (as well as the main reason the broken into version exists!)



> > It would also be more ideal if display cases could be lockpicked and vending machines hacked, as anyone who doesn't realize they need to gather money and isn't expecting to need a jackhammer is just out of luck. Those are both code projects for another day.
> > The other big problem right now with cash is having to find an ATM to load the dozens of cash cards you've looted onto a single card, which isn't ideal gameplay.
> 
> but the gun store now has an ATM. And you can carry just every cash card with you, they stack together.

As for this, most players don't even bother picking up cash cards unless they're extreme hoarders like me.

**Comment by Lamandus** [comment]
> Either way, please actually fix the layouts. I like the use of display cases but it definitely needs some sanity-checking, and for the vending machines to be removed from the normal store variants.

a pity. So no vending machines at all then.

**Comment by RoyalFox2140** [comment]
Due to NPC trading being fairly useless and normal vending machines being smashable, cash cards are mainly for making arrows. This won't change until someone re-codes a few things. Namely NPC trading, but also how cash itself works.

**Comment by RoyalFox2140** [comment]
> > You can't just upend the entire gameplay loop by putting all ammo in vending machines. That's gotta go.
> > You've also completely removed the area for a cashier to work but that's a fairly minor complaint, I'd probably move the display cases to form a cashier area because the shop looks bad without one.
> 
> grafik this is meant to be the cashier area but it isn't too clear, maybe removing the windows will help

Visibility is terrible on that, you'd have too many thieves. I'd probably use the entire top left corner as the cashier cage with a lot more display cases and to add back the computer furniture. You need to see most of the store from the cashier area to watch for thieves.

**Comment by Tefnut** [comment]
> > Either way, please actually fix the layouts. I like the use of display cases but it definitely needs some sanity-checking, and for the vending machines to be removed from the normal store variants.
> 
> a pity. So no vending machines at all then.

I don't think anyone would have had a problem if more were added, its more that they were being added as replacements to ammo cases and shelves.

One other thing i think could solve hangups is if more gun store variants (And stores in general) had storerooms where surplus ammo besides what was on display was kept, (after all, most stores carry a surplus of goods in the back) but shops as a whole are already fighting to have everything they need in their layout, you couldn't really get that concept without adding basements or second floors to more.

**Comment by Tefnut** [comment]
> > > You can't just upend the entire gameplay loop by putting all ammo in vending machines. That's gotta go.
> > > You've also completely removed the area for a cashier to work but that's a fairly minor complaint, I'd probably move the display cases to form a cashier area because the shop looks bad without one.
> > 
> > 
> > grafik this is meant to be the cashier area but it isn't too clear, maybe removing the windows will help
> 
> Visibility is terrible on that, you'd have too many thieves. I'd probably use the entire top left corner as the cashier cage with a lot more display cases and to add back the computer furniture. You need to see most of the store from the cashier area to watch for thieves.

Honestly, thats why the cashier cage in that layout was in the bottom, and was the entire bottom.

**Comment by Lamandus** [comment]
here are the improved versions of the 2 changed gunshops:

grafik

grafik

**Comment by RoyalFox2140** [comment]
I like the cashier area, this works.

**Review by chaosvolt — APPROVED** [review]
